### PR TITLE
Allow instance rename

### PR DESCRIPTION
**Pull request by MusicDin**
Instead of replacing an instance, allow instance to be renamed. This is only possible if instance is either currently stopped or the `running` field is set to `false`.

### CODE REVIEW

File: internal/instance/resource_instance_test.go [hunk 1]
```diff
@@ -216,6 +216,55 @@ func TestAccInstance_restartVirtualMachine(t *testing.T) {
...
+				Config: testAccInstance_rename(instanceNameB, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameB),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Running"),
+				),
+			},
+			{
+				// Ensure instance rename fails when instance is running.
+				Config:      testAccInstance_rename(instanceNameA, true),
+				ExpectError: regexp.MustCompile("Renaming of running instance not allowed"),
```
Comment (simondeziel) [inline internal/instance/resource_instance_test.go:262]: Should we also have the `resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameB),` checked here?

File: internal/instance/resource_instance_test.go [hunk 2]
```diff
@@ -216,6 +216,55 @@ func TestAccInstance_restartVirtualMachine(t *testing.T) {
...
+				// Launch a new instance.
+				Config: testAccInstance_rename(instanceNameA, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameA),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Running"),
+				),
+			},
+			{
+				// Stop and rename the instance.
+				Config: testAccInstance_rename(instanceNameB, false),
```
Comment (simondeziel) [inline internal/instance/resource_instance_test.go:237]: The provided name could be a bit more indicative of the intention. Say the instance started with `A`, then move to `A2B` -> `B2C` -> `C2D` -(fail)-> `D2E`.

Just an idea as I think what you have is fine, it just required a bit more mental tracking.

**Review by simondeziel — APPROVED** [review]

File: internal/instance/resource_instance_test.go (hunk 1 above)
Comment (MusicDin) [reply internal/instance/resource_instance_test.go:262]: Not sure if state is testable when error is expected, but worth trying.

File: internal/instance/resource_instance_test.go (hunk 2 above)
Comment (MusicDin) [reply internal/instance/resource_instance_test.go:237]: Hehe, will fix :)

File: internal/instance/resource_instance_test.go (hunk 1 above)
Comment (MusicDin) [reply internal/instance/resource_instance_test.go:269]: Seems to be working. Thanks :)

**Review by simondeziel — APPROVED** [review]

**Pull request merged**